File: pinyougou-web/pinyougou-user-web/src/main/webapp/js/controller/userController.js
/** 定义控制器层 */
app.controller('userController', function($scope, baseService){

    //定义用户对象
    $scope.user = {};

    //用户注册
    $scope.save = function () {
        if($scope.password && $scope.password == $scope.user.password){
            baseService.sendPost("/user/save?smsCode"+$scope.smsCode,$scope.user)
                .then(function (response) {
                    if(response.data){
                        alert("注册成功!");
                        $scope.user = {};
                        $scope.password = "";
                        $scope.smsCode = "";
                    }else {
                        alert("注册失败!")
                    }

                });
        }else{
            alert("两次密码不一致!")
        }
    }

    //发送短信验证吗到用户手机
    $scope.sendCode = function () {
        if($scope.user.phone && /^1[3|5|7|8|9]\d{9}$/.test($scope.user.phone)){
            baseService.sendGet("/user/sendCode?phone="+$scope.user.phone)
                .then(function (response) {
                    // if(response.data){
                    //     alert("发送成功!");
                    // }else {
                    //     alert("发送失败!");
                    // }
                    alert(response.data?"发送成功!":"发送失败!");
                });
        }else {
            alert("手机号码格式不正确!");
        }
    };

});

File: pinyougou-web/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.User;
import com.pinyougou.service.UserService;
import org.apache.commons.lang3.StringUtils;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * @Author: ZhaJing
 * @Date: Created in 2018/11/17 12:31
 * @Description: // TODO
 * @Version: 1.0
 */
@RestController
@RequestMapping("/user")
public class UserController {
	@Reference(timeout = 10000)
	private UserService userService;

	/**
	 * @Author ZhaJing
	 * @Description //TODO 注册用户
	 * @Date 12:32 2018/11/17
	 * @Param [user]
	 * @return
	 **/
	@RequestMapping("/save")
	public boolean save(@RequestBody User user,String smsCode){
		try {
			boolean ok = userService.checkSmsCode(user.getPhone(), smsCode);
			if(ok){
				userService.save(user);
				return true;
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
		return false;
	}

	/**
	 * @Author ZhaJing
	 * @Description //TODO 发送短信验证码
	 * @Date 21:30 2018/11/20
	 * @Param []
	 * @return
	 **/
	@GetMapping("/sendCode")
	public boolean sendCode(String phone){
		try {
			if(StringUtils.isNoneBlank(phone)){
				//发送验证码
				userService.sendCode(phone);
				return true;
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
		return false;
	}

}
